direct/nn/iterdualnet/iterdualnet.py

```diff
@@ -0,0 +1,201 @@
+# coding=utf-8
+# Copyright (c) DIRECT Contributors
+
+from typing import Callable
+
+import torch
+from torch import nn
+
+import direct.data.transforms as T
+from direct.nn.unet.unet_2d import NormUnetModel2d, UnetModel2d
+
+
+class IterDualNet(nn.Module):
+    r"""Iterative Dual Network solves iteratively the following problem
+
+    .. math ::
+
+        \min_{x} ||A(x) - y||_2^2 + \lambda_I ||x - D_I(x)||_2^2 + \lambda_F ||x - \mathcal{Q}(D_F(f))||_2^2, \quad
+        \left\{ \begin{array} Q = \mathcal{F}^{-1}, f = \mathcal{F}(x) & \text{if compute_per_coil is False} \\
+        Q = \mathcal{F}^{-1} \circ \mathcal{E}, f = \mathcal{R} \circ \mathcal{F}(x) & \text{otherwise} \end{array}
+
+    by unrolling a gradient descent scheme where :math:`\mathcal{E}` and :math:`\mathcal{R}` are the expand and
+    reduce operators which use the sensitivity maps. :math:`D_I` and :math:`D_F` are trainable U-Nets operating
+    in the image and k-space domain.
+    """
+
+    def __init__(
+        self,
+        forward_operator: Callable,
+        backward_operator: Callable,
```

Comment:
I do agree with Jonas that Callable is probably too wide. I like the idea of doing an operator class.